File: Foody/app/src/main/java/prateek_gupta/foody/data/Repository.java
package prateek_gupta.foody.data;

import javax.inject.Inject;

import dagger.hilt.android.scopes.ActivityRetainedScoped;
import dagger.hilt.android.scopes.ViewModelScoped;

@ViewModelScoped
public class Repository {
    RemoteDataSource remote;
    LocalDataSource local;

    @Inject
    public Repository(RemoteDataSource remoteDataSource,LocalDataSource localDataSource) {
        remote = remoteDataSource;
        local=localDataSource;
    }

    public RemoteDataSource getRemote() {
        return remote;
    }

    public LocalDataSource getLocal() {
        return local;
    }
}

File: Foody/app/src/main/java/prateek_gupta/foody/data/Repository.kt
package prateek_gupta.foody.data

import dagger.hilt.android.scopes.ActivityRetainedScoped
import dagger.hilt.android.scopes.ViewModelScoped
import javax.inject.Inject

@ActivityRetainedScoped
class Repository @Inject constructor(
    remoteDataSource: RemoteDataSource,
    localDataSource: LocalDataSource
) {
    val remote=remoteDataSource
    val local=localDataSource
}